vault/rollback.go

```diff
@@ -119,7 +129,42 @@ func (m *RollbackManager) run() {
 func (m *RollbackManager) triggerRollbacks() {
 	backends := m.backends()
 
-	for _, e := range backends {
+	var jobs chan func()
+	var wg sync.WaitGroup
+	firstRun := m.firstRun
+	m.firstRun = false
+	if firstRun {
+		// Use a small temporary worker pool to run the very first rollbacks in parallel, as they will trigger
+		// backend initialization
+		numWorkers := runtime.NumCPU() * 2 // For existing releases, don't modify the current behavior without an env override
+		if v := os.Getenv("VAULT_INITIAL_ROLLBACK_CONCURRENCY"); v != "" {
+			pv, err := strconv.Atoi(v)
+			if err != nil || pv < 1 {
+				m.logger.Warn("invalid value for VAULT_INITIAL_ROLLBACK_CONCURRENCY, must be a positive integer", "error", err, "value", pv)
+			} else {
+				numWorkers = pv
+			}
+		}
+		jobs = make(chan func())
+		// Start 'em
+		for i := 0; i < numWorkers; i++ {
+			go func() {
+				for v := range jobs {
+					v()
+					wg.Done()
+				}
+			}()
+		}
+
+		// Randomize the jobs on first run to allow progress on migrations, etc to be made if there are restarts/reunseals
```

Comment:
Clever.

Reply:
Yeah, but maybe unnecessary

Reply:
Are backend uuid's already randomly assigned on mount?

Reply:
No, they're persistent.

Reply:
Ah interesting, I must be thinking of a different UUID then.